connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala

```diff
@@ -292,3 +331,31 @@ class TestForeachWriter[T] extends ForeachWriter[T] {
 case class TestClass(value: Int) {
   override def toString: String = value.toString
 }
+
+class EventCollector extends StreamingQueryListener {
+  @volatile var startEvent: QueryStartedEvent = null
+  @volatile var terminationEvent: QueryTerminatedEvent = null
+  @volatile var idleEvent: QueryIdleEvent = null
+
+  private val _progressEvents = new mutable.Queue[StreamingQueryProgress]
+
+  def progressEvents: Seq[StreamingQueryProgress] = _progressEvents.synchronized {
+    _progressEvents.clone().toSeq
+  }
+
+  override def onQueryStarted(event: StreamingQueryListener.QueryStartedEvent): Unit = {
+    startEvent = event
+  }
+
+  override def onQueryProgress(event: StreamingQueryListener.QueryProgressEvent): Unit = {
+    _progressEvents += event.progress
```

Comment:
Could you update this to do a spark operation? Like spark.write() or spark.read()?

Reply:
If this not feasible now, we could do that a follow up to this PR with a TODO.

Reply:
Updated the unit test to cover a `spark.write()` case